src/components/app/details/cdDetails/CDDetails.tsx

```diff
@@ -446,23 +472,31 @@ const HistoryLogs: React.FC<{
         autoBottomScroll: triggerDetails.status.toLowerCase() !== 'succeeded',
     })
 
+    let artifactTodeploymentHistoryIndexMap = new Map<number,number>()
+    deploymentHistoryResult.forEach((val,i)=>{
```

Comment:
This should come from parent so that it gets calculated only when result changes